src/main/scala/mesosphere/util/state/FrameworkIdUtil.scala

```diff
@@ -0,0 +1,58 @@
+package mesosphere.util.state
+
+import org.apache.mesos.Protos.{ FrameworkInfo => FrameworkInfoProto, FrameworkID => FrameworkIDProto }
+import com.google.protobuf.InvalidProtocolBufferException
+import java.util.logging.{ Level, Logger }
+import scala.concurrent.duration.Duration
+import scala.util.{ Failure, Success }
+import scala.concurrent.{ Future, Await, ExecutionContext }
+
+/**
+  * Utility class for keeping track of a framework ID in Mesos state.
+  *
+  * @param state State implementation
+  * @param key The key to store the framework ID under
+  */
+
+class FrameworkIdUtil(val state: PersistentStore, val key: String = "frameworkId") {
```

Comment:
This class lives in mesos-utils and depend on the State abstraction.
I basically copied this class and made the needed adaptions.
Yes: it should also use MarathonStore - ok I will change this also.